### ISSUE TITLE
Status of Neovim and Ex integration?

### ISSUE THREAD

**Issue by mhchen** [VSCodeVim/Vim#4898]
Sorry, this doesn't fit into either of your templates as it's just a question for a potential contributor. First, the obligatory thanks for making this project: Typescript on regular vim was making me really yearn for a real IDE and this project has single-handedly let me make the switch and be somewhat productive.

Have the maintainers of this project decide whether to rely on Neovim for Ex mode going forward, or to continue maintaining the custom command line integration? The Ex mode features like `:g` and project traversal via command line are what I miss most about vim now. I forked this repo to start working on some of the existing bugs (e.g. https://github.com/VSCodeVim/Vim/issues/4730) and some new features, but I'm not sure it's worth it to work on the custom command mode if the long-term goal is to integrate more strongly with Neovim anyway.

Let me know if you have any clarity on this project's direction with this feature. Also, if there is a better forum for this kind of discussion, please let me know that too. I joined the Slack but it doesn't seem super active. I assume Github Issues are the way to go on stuff like this?

**Comment by J-Fields** [comment]
> Also, if there is a better forum for this kind of discussion, please let me know that too. I joined the Slack but it doesn't seem super active. I assume Github Issues are the way to go on stuff like this?

Like you said, the slack isn't super active. Here or there is fine for discussions like this (though I tend to check here more frequently).

There currently aren't any plans to integrate with neovim except for ex mode (https://github.com/asvetliakov/vscode-neovim is an attempt to do that). For ex mode, I accept PRs improving the non-neovim experience, but personally don't prioritize issues that work with neovim. I'm not sure what the endgame is tbh, but if you want to implement `:g`, for instance, I know people would definitely appreciate it!

**Comment by mhchen** [comment]
Makes sense. The things I'd prioritize are things like better tab-autocomplete for file paths and things like `expand()`, or even commands from https://github.com/tpope/vim-eunuch. What do you think @J-Fields? Would you be open to such improvements?

**Comment by J-Fields** [comment]
> Would you be open to such improvements?

Almost definitely

**Comment by J-Fields** [comment]
A quick update to this: recently I've been sinking a lot of time into re-writing a lot of vimscript/ex-command stuff to support more complex use cases. I think the endgame will actually be removing neovim integration altogether, but it'll be a while before we're ready to do that.
